### ISSUE TITLE
support ctrl/shift+up/down for moving commits around (a-la sublime-text)

### ISSUE THREAD

**Issue by koreno** [MitMaro/git-interactive-rebase-tool#233]
getting inspiration from sublime-text:

- shift + up/down - add to commit selection (for moving multiple commits together)
- ctrl+shift + up/down - move commit the selection (instead of j/k, which require more finger hopping...)

**Comment by MitMaro** [comment]
Thanks for the request!

Before adding the visual mode, I experimented with using `shift+up/down` for selecting a set of commits. Sadly it's not supported universally across operating systems and/or keyboards. Different operating systems and even certain hardware implementations do not report the same value for shift+up/down.

I can however add the ability for the `shift+up/down` keybinding to be used, for those who wish to customize their key bindings and have a system that supports it. But it won't be usable to replace the visual mode.

I would also like to add `ctrl+up/down` to move selected commits as it is closer to what I use in editors I used for moving lines. However, there is currently no way that I am aware of to receive this event from the Curses libraries. For example, on my system when I press `ctrl+up` it actually reports as a `6` and `ctrl+down` reports as a `!`. Adding `shift+ctrl+up/down` doesn't report any input, at least on my laptop and `shift+ctrl+left` reports as `\"` and `shift+ctrl+right` reports as a `1`.
